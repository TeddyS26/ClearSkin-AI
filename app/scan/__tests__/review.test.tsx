import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Review from "../review";
import { useRouter, useLocalSearchParams } from "expo-router";

jest.mock("expo-router");
jest.mock("lucide-react-native", () => ({
  CheckCircle: "CheckCircle",
  RotateCcw: "RotateCcw",
  Sparkles: "Sparkles",
  ArrowLeft: "ArrowLeft",
}));

describe("Review", () => {
  const mockRouter = { push: jest.fn(), back: jest.fn() };
  const mockParams = {
    front: "file:///front.jpg",
    left: "file:///left.jpg",
    right: "file:///right.jpg",
  };

  beforeEach(() => {
    jest.clearAllMocks();
    (useRouter as jest.Mock).mockReturnValue(mockRouter);
    (useLocalSearchParams as jest.Mock).mockReturnValue(mockParams);
  });

  it("should render review screen", () => {
    const { getByText } = render(<Review />);
    
    expect(getByText("Review Photos")).toBeTruthy();
    expect(getByText("Make sure all photos are clear and well-lit")).toBeTruthy();
  });

  it("should display all three photo labels", () => {
    const { getByText } = render(<Review />);
    
    expect(getByText("Front View")).toBeTruthy();
    expect(getByText("Left View")).toBeTruthy();
    expect(getByText("Right View")).toBeTruthy();
  });

  it("should render start analysis button", () => {
    const { getByText } = render(<Review />);
    
    expect(getByText("Start Analysis")).toBeTruthy();
  });

  it("should render retake photos button", () => {
    const { getByText } = render(<Review />);
    
    expect(getByText("Retake Photos")).toBeTruthy();
  });

  it("should navigate to loading when start analysis pressed", () => {
    const { getByText } = render(<Review />);
    
    fireEvent.press(getByText("Start Analysis"));
    
    expect(mockRouter.push).toHaveBeenCalledWith({
      pathname: "/scan/loading",
      params: mockParams,
    });
  });

  it("should go back when retake photos pressed", () => {
    const { getByText } = render(<Review />);
    
    fireEvent.press(getByText("Retake Photos"));
    
    expect(mockRouter.back).toHaveBeenCalled();
  });

  it("should navigate back to capture when back button pressed", () => {
    const { getAllByText } = render(<Review />);
    
    const backButton = getAllByText("Back")[0];
    fireEvent.press(backButton);
    
    expect(mockRouter.push).toHaveBeenCalledWith("/scan/capture");
  });

  it("should display images with correct URIs", () => {
    const { toJSON } = render(<Review />);
    const tree = toJSON();
    
    expect(tree).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { toJSON } = render(<Review />);
    
    expect(toJSON()).toMatchSnapshot();
  });
});

