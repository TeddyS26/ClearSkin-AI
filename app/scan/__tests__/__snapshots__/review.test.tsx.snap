// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should match snapshot 1`] = `
<RNCSafeAreaView
  className="flex-1 bg-emerald-50"
  edges={
    {
      "bottom": "off",
      "left": "off",
      "right": "off",
      "top": "additive",
    }
  }
>
  <RCTScrollView
    className="flex-1"
    contentContainerStyle={
      {
        "paddingBottom": 100,
      }
    }
  >
    <View>
      <View
        className="px-6 pt-6"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="flex-row items-center mb-4 active:opacity-60"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <ArrowLeft
            color="#10B981"
            size={24}
            strokeWidth={2.5}
          />
          <Text
            className="text-emerald-600 font-semibold text-base ml-1"
          >
            Back
          </Text>
        </View>
        <View
          className="mb-6"
        >
          <Text
            className="text-2xl font-bold text-gray-900 mb-2"
          >
            Review Photos
          </Text>
          <Text
            className="text-base text-gray-600"
          >
            Make sure all photos are clear and well-lit
          </Text>
        </View>
        <View
          className="mb-4"
        >
          <View
            className="flex-row items-center mb-2"
          >
            <View
              style={
                {
                  "marginRight": 8,
                }
              }
            >
              <CheckCircle
                color="#10B981"
                size={18}
                strokeWidth={2.5}
              />
            </View>
            <Text
              className="text-base font-semibold text-gray-900"
            >
              Front View
            </Text>
          </View>
          <View
            className="bg-white rounded-3xl overflow-hidden shadow-sm border border-gray-100"
          >
            <Image
              resizeMode="cover"
              source={
                {
                  "uri": "file:///front.jpg",
                }
              }
              style={
                {
                  "height": 240,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          className="mb-4"
        >
          <View
            className="flex-row items-center mb-2"
          >
            <View
              style={
                {
                  "marginRight": 8,
                }
              }
            >
              <CheckCircle
                color="#10B981"
                size={18}
                strokeWidth={2.5}
              />
            </View>
            <Text
              className="text-base font-semibold text-gray-900"
            >
              Left View
            </Text>
          </View>
          <View
            className="bg-white rounded-3xl overflow-hidden shadow-sm border border-gray-100"
          >
            <Image
              resizeMode="cover"
              source={
                {
                  "uri": "file:///left.jpg",
                }
              }
              style={
                {
                  "height": 240,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          className="mb-4"
        >
          <View
            className="flex-row items-center mb-2"
          >
            <View
              style={
                {
                  "marginRight": 8,
                }
              }
            >
              <CheckCircle
                color="#10B981"
                size={18}
                strokeWidth={2.5}
              />
            </View>
            <Text
              className="text-base font-semibold text-gray-900"
            >
              Right View
            </Text>
          </View>
          <View
            className="bg-white rounded-3xl overflow-hidden shadow-sm border border-gray-100"
          >
            <Image
              resizeMode="cover"
              source={
                {
                  "uri": "file:///right.jpg",
                }
              }
              style={
                {
                  "height": 240,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="bg-emerald-500 py-5 rounded-2xl items-center mb-4 shadow-sm flex-row justify-center active:opacity-90"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              {
                "marginRight": 8,
              }
            }
          >
            <Sparkles
              color="#FFFFFF"
              size={22}
              strokeWidth={2}
            />
          </View>
          <Text
            className="text-white font-bold text-lg"
          >
            Start Analysis
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          className="bg-white border-2 border-gray-200 py-5 rounded-2xl items-center flex-row justify-center active:opacity-90"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              {
                "marginRight": 8,
              }
            }
          >
            <RotateCcw
              color="#6B7280"
              size={20}
            />
          </View>
          <Text
            className="text-gray-700 font-semibold text-base"
          >
            Retake Photos
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
